### ISSUE TITLE
Investigate fill_probability capped at 0.3 across all items

### ISSUE THREAD

**Issue by MarcusFranz** [MarcusFranz/gept#14]
## Problem

The `fill_probability` column in the `predictions` table is hard-capped at **0.3** for every item, every `hour_offset`, and every `offset_pct`. No prediction row in the entire table exceeds 0.3.

```sql
SELECT MAX(fill_probability) FROM predictions 
WHERE time = (SELECT MAX(time) FROM predictions);
-- Result: 0.3
```

The values approach 0.3 asymptotically (e.g. 0.29998, 0.29997) but never exceed it, suggesting the model output is being clipped or the sigmoid calibration is saturating.

## Impact

- The price drop monitor originally filtered on `fill_probability >= 0.5`, which was **impossible to satisfy** — effectively making the alerting system dead code since launch.
- Workaround deployed: monitor now uses conservative `offset_pct` selection instead of filtering on fill probability. But the column should still produce meaningful probabilities for other uses (ranking, confidence display).

## Investigation areas

1. **Model output layer** — Is there a sigmoid/softmax that saturates? Check the final activation and whether the training targets are scaled correctly.
2. **Calibration** — Was the model trained on binary fill/no-fill labels? If fill rates in training data are very low, the model may learn to never predict > 0.3.
3. **Post-processing** — Check if `fill_probability` is being clipped or capped anywhere in the inference pipeline before writing to the predictions table.
4. **Training data balance** — What fraction of training samples actually filled? If < 30%, the model might be correctly calibrated but on imbalanced data.

## Relevant code

- Inference pipeline: `packages/model/src/pipeline/run_patchtst_inference.py`
- Predictions table write: check where `fill_probability` is computed and inserted
- Price monitor (consumer): `packages/engine/src/trade_price_monitor.py`

## Labels

bug, model

**Comment by MarcusFranz** [comment]
## Resolution: Working as Designed

Investigation confirms the 0.3 cap on `fill_probability` is **intentional**, not a bug.

### Root Cause
The cap is a deliberate safety mechanism defined in `packages/model/src/inference_config.py`:

```python
MAX_PROBABILITY = 0.30  # predictions above this are broken
CALIBRATED_MAX = 0.25   # calibration output cap (leaves headroom)
```

### Why It Exists
Backtest calibration analysis (`docs/archived-training/packages/model/docs/calibration_analysis.md`) showed:

| Bucket | Predicted | Actual | Error | Status |
|--------|-----------|--------|-------|--------|
| 0-30% | Conservative | Higher than predicted | Safe ✅ |
| **30-50%** | **34.46%** | **6.25%** | **-28.21%** | **⚠️ BROKEN** |

Only 80 predictions (0.02% of 393,578 total) fell in the 30-50% range — classic overfitting on rare patterns that don't generalize. Capping at 0.30 prevents these catastrophically miscalibrated predictions from reaching users.

### The Original Problem
The price drop monitor filtering on `fill_probability >= 0.5` was indeed dead code, but this was a consumer-side bug (unreachable threshold), not a model bug. The workaround using `offset_pct` selection is the correct approach.

### Architecture
1. Raw model output → `predict_proba()` (0-1 range)
2. Calibration layer → isotonic regression caps at 0.25
3. Hard clip → `clip_probability()` enforces [0.001, 0.30]
4. Database write → predictions table

No code changes needed. Closing as designed.
